### PR TITLE
Feature: Reset password with email implementation

### LINKED ISSUE #89: Feature:Forgot password
This issue deals with the implementation of forgot password functionality

where a person will be able to reset his password if he or she has forgotten it

### PR DESCRIPTION
**Pull request by Latiah**
**What does this PR do?**
This PR includes the implementation of the reset password for registered users where an otp is sent to their  email s so that it can be used for reseting the password.

**Description of Task to be completed?**
Modify the email template on supabase 
make the select method screen display email 
increase the number of input fields for the otp as supabase was providing 6 digits

**How should this be manually tested?**
Navigate to the sign in page and enter your email then click on the forgot password and you will be directed to a screen where you will chose the method you want to use to reset your password  and from this screen you will be directed to fill in the otp which is the one you received on your email after filling it in you will go to a screen of setting a new password and after entering your new password and confirming it a modal will pop up confirming that your password is set and then you will be redirected to the home screen 
 
**Any background context you want to provide?**
First sign up before using your email to reset the password because if it is not registered you will not receive the token you can use to reset your password on that email

What are the relevant pivotal tracker/Trello stories?
N/A

### CODE REVIEW

**Review by 911samuel — APPROVED** [review]
Amazing work

**Review by twizerim — APPROVED** [review]

**Comment by Adadie** [comment]
Hey @Latiah I'm unable to add an email I want to remember the password for. Find below a screenshot of what I'm seeing:
Screenshot_20240612_174441_Expo Go

**Comment by Latiah** [comment]
> Hey @Latiah I'm unable to add an email I want to remember the password for. Find below a screenshot of what I'm seeing: Screenshot_20240612_174441_Expo Go

you had to enter it on the sign in screen

**Comment by Latiah** [comment]
> > Hey @Latiah I'm unable to add an email I want to remember the password for. Find below a screenshot of what I'm seeing: Screenshot_20240612_174441_Expo Go
> 
> you had to enter it on the sign in screen

Also first register cause I used a different project if you registered on Sam's that does not count

File: app/reset-password/index.tsx
Comment (Emmyfrank) [inline app/reset-password/index.tsx:1]: i think you can try to work on these three simple changes
when you click on email field , the keyboard comes up and hide the email field and is hard to see what you are typing
# before your type your email
Screenshot 2024-06-12 at 21 03 08
#after when you start typing your email
Screenshot 2024-06-12 at 21 03 24
# I am also getting this network error on my end while my network is strong
Screenshot 2024-06-12 at 21 03 49

**Review by Emmyfrank — APPROVED** [review]
nice work, you have worked on my comments

**Comment by Latiah** [comment]
@Adadie I worked on the comments you gave.

**Review by Adadie — CHANGES_REQUESTED** [review]
Hey @Latiah awesome work on the implementation. Here's a few things you improve to make it much better:

1. Reduce the resend otp link time to not more than 2 minutes. It doesn't provide a good user experience for the user to wait for a whole hour to be able to resend
2. Add loading states across all the functionalities. Also, when a request is loading I should not be able to submit another request i.e I received two OTP emails because I clicked the button multiple times.
3. When I've changed my password I should not be able to go back to that screen for changing password. The auth session should also be clear, I was able to change my password multiple times by going back at that screen.

Let me know if all this makes sense to you

**Comment by Latiah** [comment]
> Hey @Latiah awesome work on the implementation. Here's a few things you improve to make it much better:
> 
> 1. Reduce the resend otp link time to not more than 2 minutes. It doesn't provide a good user experience for the user to wait for a whole hour to be able to resend
> 2. Add loading states across all the functionalities. Also, when a request is loading I should not be able to submit another request i.e I received two OTP emails because I clicked the button multiple times.
> 3. When I've changed my password I should not be able to go back to that screen for changing password. The auth session should also be clear, I was able to change my password multiple times by going back at that screen.
> 
> Let me know if all this makes sense to you

Let me work on them very quickly

**Comment by Latiah** [comment]
@Adadie  I worked on these comments you provided you can check again.

**Pull request merged**